src/Microsoft.Identity.Client/Config/ConfidentialClientApplicationBuilder.cs

```diff
@@ -0,0 +1,157 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Identity.Client.Config
+{
+#if !ANDROID_BUILDTIME && !iOS_BUILDTIME && !WINDOWS_APP_BUILDTIME // Hide confidential client on mobile platforms
+
+    /// <summary>
+    /// </summary>
+    public class ConfidentialClientApplicationBuilder : AbstractConfigurationBuilder<ConfidentialClientApplicationBuilder>
+    {
+        /// <inheritdoc />
+        internal ConfidentialClientApplicationBuilder(ApplicationConfiguration configuration)
+            : base(configuration)
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public static ConfidentialClientApplicationBuilder CreateWithApplicationOptions(ApplicationOptions options)
```

Comment:
Shouldn't it be from ConfidentialApplicationOptions ?
(with ConfidentialApplicationOptions deriving from ApplicationOptions, of course, which it does not today)

It seems today you'd propose we do:

```CSharp
ApplicationOptions options =ReadApplicationOptionsFromJSon("appsettings.json");
ConfidentialApplicationOptions options2 =ReadConfidentialApplicationOptionsFromJSon("appsettings.json");// ?? same file? 
 ConfidentialClientApplicationBuilder builder = ConfidentialClientApplicationBuilder.CreateWithApplicationOptions(options)
.WithConfidentialApplicationOptions(options2 )
.WithClientCredential(clientCredentials)

IConfidentialClientApplication app = builder.Build();
```
In reality, options2 and options should be the same as they will most probably come from the configuration framework?